test/utils/volume/vol_utils.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package volutil
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2elog "k8s.io/kubernetes/test/e2e/framework/log"
+)
+
+func CountAllVolumeAttachments(cs clientset.Interface) int {
+	e2elog.Logf("Getting all volume attachments")
+	vatts, err := cs.StorageV1().VolumeAttachments().List(context.TODO(), metav1.ListOptions{})
+	if !errors.IsNotFound(err) {
+		framework.ExpectNoError(err)
+	}
+	return len(vatts.Items)
+}
+
+// waitForPersistentVolumeClaimDeleted waits for a PersistentVolumeClaim to be removed from the system until timeout occurs, whichever comes first.
+func WaitForPersistentVolumeClaimDeleted(c clientset.Interface, ns string, pvcName string, Poll, timeout time.Duration) error {
+	framework.Logf("Waiting up to %v for PersistentVolumeClaim %s to be removed", timeout, pvcName)
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(Poll) {
```

Comment:
We should use `wait.PollImmediate` instead of implementing our own polling loop.

Reply:
I don't think this function was implemented for the scale tests. I think Akash has just refactored it. It is from this PR: https://github.com/kubernetes-sigs/azuredisk-csi-driver/pull/428
Would certainly be good to address it.

Reply:
And actually, since this function is not used anywhere besides test/types/resource_types.go we should probably leave it there instead of refactoring it to the utils.

Reply:
While sorting out these helper functions, I had to resolve a whole lot of circular dependencies to split these functions into packages. I would really not want to keep functions which are not intended by their purpose in a certain package.

@edreed Will implement the for loop with PollImmediate, since it doesnt look like that would create any dependency issues.

Reply:
@mukhoakash I have addressed this already.